emperor/flat_plots.py

```diff
@@ -0,0 +1,352 @@
+        plot = two_dimensional_gradient
+        legend = legend_gradient
+    else:
+        plot = two_dimensional_discrete
+        legend = legend_discrete
+
+    # full page, with square plots
+    fig, grid = plt.subplots(3, 4, sharex=False, sharey=False,
+                             figsize=(24, 16))
+    grid[0, 3].grid(False)
+    grid[0, 3].set_axis_off()
+    grid[2, 3].grid(False)
+    grid[2, 3].set_axis_off()
+
+    for row in range(3):
+        # y_pc is the axis in the ordination to put on the y-axis
+        y_pc = row + offset
+
+        # the y-axis is consistent across a row so only label the
+        # first column
+        row_ax = grid[row, 0]
+        row_ax.set_ylabel(_pcoa_label(y_pc, ordination), fontsize=14)
+
+        for col in range(3):
+            # x_pc is the axis in the ordination to put on the x-axis
+            x_pc = row + col + offset + 1
+
+            # fetch the specific subplot, and plot
+            ax = grid[row, col]
+            ax.set_xlabel(_pcoa_label(x_pc, ordination), fontsize=14)
+            plot(ax, field, mapping, ordination, x_pc, y_pc, colors, null_kw,
+                 focus_kw)
+
+    # establish the legend on the right side
+    legend(grid[1, 3], field, mapping, ordination, colors, null_kw)
+
+    return (fig, grid)
+
+
+def _pcoa_label(dim, ordination):
+    """Helper method to make a PCoA label"""
+    fmt = 'PC%d (%0.2f%%)'
+    return fmt % (dim + 1, ordination.proportion_explained[dim] * 100)
+
+
+def two_dimensional_gradient(ax, field, mapping, ordination, xaxis, yaxis,
+                             colormap, null_kw, focus_kw):
+    """Plot two axes
+
+    Parameters
+    ----------
+    ax : AxesSubplot
+        A subplot to plot into
+    field : str
+        A metadata field in the mapping to plot.
+    mapping : pd.DataFrame
+        The metadata about the samples.
+    ordination : skbio.OrdinationResults
+        The ordination object.
+    xaxis : integer
+        An axis to plot from the ordination.
+    yaxis : integer
+        An axis to plot from the ordination.
+    colormap : str
+        The colormap to use.
+    null_kw : dict
+        Additional arguments to the scatter plot for the samples in the
+        background.
+    focus_kw : dict
+        Additional arguments to the scatter plot for the samples in focus.
+    """
+    gap_to_frame = 0.01
+
+    mapping = mapping.loc[ordination.samples.index]
+    field_values = mapping[~mapping[field].isnull()][field]
+
+    nonfield_samples = set(mapping.index) - set(field_values.index)
+
+    # discretize our color gradient, and determine which bin (i.e., color)
+    # a given value is associated with
+    bins = np.arange(field_values.min(),
+                     field_values.max(),
+                     (field_values.max() - field_values.min()) / 256.0)
+
+    order = np.digitize(field_values, bins=bins, right=True)
+    colors = np.array(sns.color_palette(colormap, n_colors=len(bins) + 1))
```

Comment:
this can be done as follows
```
    cNorm = colors.Normalize(vmin=field_values.min(), vmax=field_values.max())
    scalarMap = cmx.ScalarMappable(norm=cNorm, cmap=colormap)
    colors = scalarMap.to_rgba(np.linspace(0, 1, 256))
```